app/components/modals/modal-base.js

```diff
@@ -15,6 +15,9 @@ export default UiModal.extend({
     if (this.isOpen) {
       $element.modal(this.defaultOptions).modal('show');
     } else {
+      if(this.unloadRecord){
```

Comment:
Please remove. You already have a callback for modal closing. Makes no sense to put logic of user role modal in modal base

Reply:
@iamareebjamal then the solution to this problem could be disabling clicking outside the window so that the modal won't unless user click on cancel button and clicking on cancel button is already deleting the empty record created.

Reply:
Not really. There are tons of ways to solve it

Reply:
@iamareebjamal So in the previous solution where I was calling the function when the user click outside the box in the modal-base.Is this not be good enough sol? bcz what actually we want to do is that calling the function when the modal closes.

Reply:
> Please remove. You already have a callback for modal closing. Makes no sense to put logic of user role modal in modal base

@iamareebjamal and I have removed the function which was calling for modal closing and put it inside modal-base bcz we want same thing to happen even if user click on cancel button or outside the model